changelog.d/3596.bugfix

```diff
@@ -0,0 +1 @@
+Respond with M_NOT_FOUND when profiles are not found locally or over federation. Fixes #3585
```

Comment:
@Half-Shot ideally, this would be `3585.bugfix`, as the news file should point to the bug where one is available, and the bug should reference the fix (which github does automatically if you say the PR "fixes" a ticket)

Reply:
Sure, want me to fix?